connectors/src/connectors/zendesk/lib/help_center_permissions.ts

```diff
@@ -25,6 +25,170 @@ import {
+    }
+  }
+
+  await changeZendeskClientSubdomain({ client: zendeskApiClient, brandId });
+  try {
+    const categories = await zendeskApiClient.helpcenter.categories.list();
+    categories.forEach((category) =>
+      allowSyncZendeskCategory({
+        subdomain,
+        connectionId,
+        connectorId,
+        categoryId: category.id,
+      })
+    );
+  } catch (e) {
+    logger.error(
+      { connectorId, brandId },
+      "[Zendesk] Could not fetch categories."
+    );
+  }
+
+  return brand;
+}
+
+/**
+ * Mark a help center as permission "none" and all children (collections and articles).
+ */
+export async function revokeSyncZendeskHelpCenter({
+  connectorId,
+  brandId,
+}: {
+  connectorId: ModelId;
+  brandId: number;
+}): Promise<ZendeskBrandResource | null> {
+  const brand = await ZendeskBrandResource.fetchByBrandId({
+    connectorId,
+    brandId,
+  });
+  if (!brand) {
+    logger.error(
+      { brandId },
+      "[Zendesk] Brand not found, could not revoke sync."
+    );
+    return null;
+  }
+
+  await brand.revokeHelpCenterPermissions();
+  return brand;
+}
+
+export async function allowSyncZendeskCategory({
+  subdomain,
+  connectorId,
+  connectionId,
+  categoryId,
+}: {
+  subdomain: string;
+  connectorId: ModelId;
+  connectionId: string;
+  categoryId: number;
+}): Promise<ZendeskCategoryResource> {
+  let category = await ZendeskCategoryResource.fetchByCategoryId({
+    connectorId,
+    categoryId,
+  });
+  if (category?.permission === "none") {
+    await category.update({ permission: "read" });
+  }
+
+  const token = await getZendeskAccessToken(connectionId);
+  const zendeskApiClient = createZendeskClient({ token, subdomain });
+
+  if (!category) {
+    const { result: fetchedCategory } =
+      await zendeskApiClient.helpcenter.categories.show(categoryId);
+    if (fetchedCategory) {
+      category = await ZendeskCategoryResource.makeNew({
+        blob: {
+          connectorId: connectorId,
+          brandId: fetchedCategory.id,
+          name: fetchedCategory.name || "Brand",
+          categoryId: fetchedCategory.id,
+          permission: "read",
+          url: fetchedCategory.url,
+        },
+      });
+    } else {
+      logger.error({ categoryId }, "[Zendesk] Category could not be fetched.");
+      throw new Error("Category could not be fetched.");
+    }
+  }
+
+  return category;
+}
+
+/**
+ * Mark a category with "none" permissions alongside all its children articles.
+ */
+export async function revokeSyncZendeskCategory({
+  connectorId,
+  categoryId,
+}: {
+  connectorId: ModelId;
+  categoryId: number;
+}): Promise<ZendeskCategoryResource | null> {
+  const category = await ZendeskCategoryResource.fetchByCategoryId({
+    connectorId,
+    categoryId: categoryId,
+  });
+  if (!category) {
+    logger.error(
+      { categoryId: categoryId },
+      "[Zendesk] Category not found, could not revoke sync."
+    );
+    return null;
```

Comment:
you don't want to throw for this one ?

Reply:
figured that if we were to revoke permissions for smth that didn't exist anymore we should carry on and try to process the other permissions

Reply:
I do realize that's not consistent with the behavior when allowing permissions, I'm a bit undecided on that one 🤷

Reply:
Fair !